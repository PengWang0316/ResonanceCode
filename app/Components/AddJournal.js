import React, { Component } from "react";
import Util from "../apis/Util";
import Loading from "./Loading"
import JournalContent from "./JournalContent";
import ReadingSearchAndList from "./ReadingSearchAndList";
import { isLogin } from "../apis/LoginApi";
import DatabaseApi from "../apis/DatabaseApi";
import PropTypes from "prop-types";

class AddJournal extends Component {

  componentWillMount(){
    // Initial variables
    this.readings = this.props.journalData ? this.props.journalData.readingIds : {}; // keep which readings should be attached on. Format is like {readingId: null}
    this.contents = {}; // Keep content keys and content
    this.contentKeyIndex = 0; // Generate keys for different contents
    this.contentIndexs = {}; // Use to track the index of content component in the array. Delete function needs it. The format is {contentKey: index}
    this.jounalId = this.props.journalData ? this.props.journalData._id : null; // Keeping journal id for update
    this.userId = isLogin(document).userid;

    this.state={
      isWriting: false,
      journalDate: this.props.journalData ? Util.getDateString(this.props.journalData.date) : Util.getCurrentDateString(),
      isDateCorrect: true,
      isEmptyReading: this.props.journalData ? false : true,
      contentComponentArray: [], // keep content component
      addJournalContent: "overview",
      pingPongState: this.props.journalData ? this.props.journalData.ping_pong_state : "Neutral"
    };

    console.log("Journal before alter: ", this.props.journalData);
    // If journal data exsit, get content
    if(this.props.journalData){
      // this.oldContentKeys = []; // keep original content keys
      this.oldReadingIds = Object.keys(this.readings); // keep original reading ids

      // delete unnecessary properties from journal object
      let journal = this.props.journalData;
      this.journalUserId = journal.user_id;
      delete journal.date;
      delete journal.user_id;
      delete journal.ping_pong_state;
      delete journal._id;
      delete journal.readingIds;
      let keyRegExp = /-(\d+)$/; // The regular expression for subtract suffixs
      console.log("Journal after alter: ", this.props.journalData);
      Object.keys(journal).map((key)=>{
        // this.oldContentKeys.push(key); // keeping the exsit keys in an array for delete function
        // handleAddContentClick(addJournalContent, newContentName, newContentKey, contentKeyIndex, isShared)
        // Also have to save the largest key number and use it to countine
        // journalData property is like {'overview-0': 'test overview', 'overview-0-isShared': true}
        // if key is isShared do not do anything
        let matchResult = key.match(keyRegExp);
        if (matchResult){
          if (this.contentKeyIndex < matchResult[1]) this.contentKeyIndex = matchResult[1]; // keep the max index number and user can contiune add new content
          this.handleAddContentClick(journal[key], key.replace(keyRegExp, ""), key, matchResult[1], journal[`${key}-isShared`]);
          // console.log("isShared: ",journal[`${key}-isShared`]);
        }
      });
      this.contentKeyIndex++;
    }

  }

  handleChange(event, element){
    let newState={};
    newState[element]=event.target.value;
    if(element=="journalDate") newState.isDateCorrect = Util.matchDateFormat(newState[element]);
    this.setState(newState);
  }

  handleChangeCallback(contentKey, contentText){
    console.log(contentKey, contentText);
    // update content in state
    this.contents[contentKey]=contentText;
  }

  handleDeleteContentCallback(contentKey){
    // Remove content from state
    if(this.contents.hasOwnProperty(contentKey)){
      delete this.contents[contentKey];
      delete this.contents[`${contentKey}-isShared`];
    }
    // Remove content component
    delete this.contentIndexs[contentKey];
    this.setComponentToStateArray();
    // this.state.contentComponentArray.splice(this.contentIndexs[contentKey],1);
    // this.setState({contentComponentArray: this.state.contentComponentArray});
  }

  handleSharedBoxChangeCallback(contentKey, isShared){
    this.contents[`${contentKey}-isShared`] = isShared;
  }

  handleAddContentClick(addJournalContent, newContentName, newContentKey, contentKeyIndex, isShared){
    // put component in array in order to show
    newContentKey = newContentKey ? newContentKey : `${this.state.addJournalContent}-${this.contentKeyIndex}`;
    // put the component in the track object
    this.contentIndexs[newContentKey]=<JournalContent key={contentKeyIndex ? contentKeyIndex : this.contentKeyIndex++} newContent={addJournalContent ? addJournalContent : ""} newContentName={newContentName ? newContentName : this.state.addJournalContent} newContentKey={newContentKey} handleChangeCallback={(contentKey, contentText)=>{this.handleChangeCallback(contentKey, contentText);}} handleDeleteContentCallback={(contentKey)=>{this.handleDeleteContentCallback(contentKey);}} handleSharedBoxChangeCallback={(contentKey, isShared)=>{this.handleSharedBoxChangeCallback(contentKey, isShared)}} isShared={isShared ? true : false} />;
    this.setComponentToStateArray();
    // this.setState({contentComponentArray: this.state.contentComponentArray});

    // also have to put content title to state in order to track content and update
    this.contents[newContentKey] = addJournalContent ? addJournalContent : "";
    this.contents[`${newContentKey}-isShared`] = isShared ? true : false;
  }

  setComponentToStateArray(){
    // transfor object to array in order to display
    this.state.contentComponentArray=[];
    Object.keys(this.contentIndexs).map((element)=>{
      this.state.contentComponentArray.push(this.contentIndexs[element]);
    });
    this.setState({contentComponentArray: this.state.contentComponentArray});
  }

  handleSubmit(event, isUpdate){
    event.preventDefault();
    // console.log("submit: contents:",this.contents);
    console.log("journal date:",this.state.journalDate);
    // Assemble a journal object for save
    let readingIdArray = [];
    Object.keys(this.readings).map((key)=>{readingIdArray.push(key)});
    let journal = Object.assign({_id:this.jounalId, date: new Date(this.state.journalDate), ping_pong_state: this.state.pingPongState, user_id: this.userId, readingIds: readingIdArray}, this.contents);

    if(isUpdate){
      console.log("submit: journal:", journal);
      // console.log("oldContentKeys:", this.oldContentKeys);
      console.log("oldReadingKeys:", this.oldReadingIds);
      // Assemble two arrays for deletion contents and reading
      let deleteContents = [];
      let deleteReadingIds = [];
      // this.oldContentKeys.map((element)=>{if(!this.contents.hasOwnProperty(element)) deleteContents.push(element);});
      this.oldReadingIds.map((element)=>{if(!this.readings.hasOwnProperty(element)) deleteReadingIds.push(element);});
      // journal.deleteContents = deleteContents; //todo delete deleteContents and oldContentKeys
      journal.deleteReadingIds = deleteReadingIds;
      console.log("delete contents: ",deleteContents);
      console.log("delete reading ids : ",deleteReadingIds);
      DatabaseApi.updateJournal(journal).then((result)=>{
        this.props.history.push("/reading");
      });
    }else{
      DatabaseApi.createJournal(journal).then((result)=>{
        this.props.history.push("/reading");
      });
    }
    /**/
    this.setState({isWriting: true});
    console.log("submit: journal:", journal);
  }

  /* For ReadingSearchAndList callback */
  handleAttachReadingCallback(readingId){
    this.readings[readingId] = null;
    this.setState({isEmptyReading: false});
  }

  handleDetachAttachReadingCallback(readingId){
    delete this.readings[readingId];
    this.setState({isEmptyReading: Object.keys(this.readings).length===0});
  }

  handleCancel(){
    this.props.history.push("/reading");
  }

  handleDelete(){
    console.log("Delete journal!");
  }

  render(){
    return(
      <div className="addReadingDiv">
        {this.state.isWriting && <Loading text="Creating" />}

        <div className="titleDiv">{this.props.journalData ? "Update journal for readings" : "Add a new journal for readings"}</div>
        <form className="form-horizontal" onSubmit={(event) => {this.handleSubmit(event, this.props.journalData);}}>

          <div className="text-right bottom-btn-div">
            {(!this.props.journalData || this.userId == this.journalUserId) &&
              <button type="submit" className="btn btn-info loginButton" disabled={this.state.isWriting || !(this.state.journalDate.length>0) || !(this.state.isDateCorrect) || this.state.isEmptyReading}>{this.props.journalData ? "Update" : "Submit"}</button>
            }
            {(this.props.journalData && this.userId == this.journalUserId) &&
              <button onClick={()=>{this.handleDelete()}} type="button" className="btn btn-danger loginButton">Delete</button>
            }
            <button onClick={()=>{this.handleCancel()}} type="button" className="btn btn-nomal loginButton">Cancel</button>
          </div>

            <div className="form-group row form-div">
              <label htmlFor="journalDate" className="col-sm-1 col-form-label">Date</label>
              <div className="col-sm-4">

                <input className={this.state.isDateCorrect?"form-control":"form-control form-control-warning"} type="text" placeholder="mm/dd/yyyy" id="journalDate" value={this.state.journalDate} onChange={(event)=>{this.handleChange(event, "journalDate")}} />
                {!this.state.isDateCorrect && <span className="glyphicon glyphicon-warning-sign form-control-feedback form-control-warning-span"></span>}

              </div>
              <label htmlFor="pingPongState" className="col-sm-3 col-form-label">Ping Pong State</label>
              <div className="col-sm-4">
                <select value={this.state.pingPongState} id="pingPongState" className="form-control" onChange={(event)=>{this.handleChange(event,"pingPongState");}}>
                  <option value="Neutral">Neutral</option>
                  <option value="Reading/Call">Reading/Call</option>
                  <option value="Impact absorbing">Impact absorbing</option>
                  <option value="Impact complete">Impact complete</option>
                  <option value="Response">Response</option>
                  <option value="Outward moving">Outward moving</option>
                  <option value="Outward complete">Outward complete</option>
                </select>
              </div>
            </div>

            {/* New content goes here */}
            {this.state.contentComponentArray}

            {/* Add content button and drop list */}
            <div className="row addJournalContentDiv">

              <div onClick={()=>{this.handleAddContentClick();}} className="addJournalContentBtnDiv col-sm-6"><i className="fa fa-plus-square" /> Add one content for your journal</div>
              <div className="col-sm-6">
                <select  className="form-control" onChange={(event)=>{this.handleChange(event,"addJournalContent");}}>
                  <option value="overview">Overview</option>
                  <option value="internal_environment">Internal Environment</option>
                  <option value="relational_environment">Relational Environment</option>
                  <option value="physical_environment">Physical Environment</option>
                  <option value="creative_vector">Creative Vector</option>
                  <option value="synchronicities">Synchronicities</option>
                  <option value="dreams">Dreams</option>
                  <option value="other">Other</option>
                </select>

              </div>

            </div>


            {/*  Start reading search function */}
            <ReadingSearchAndList readings={this.props.journalData ? this.readings : null} attachReadingCallback={(readingId)=>{this.handleAttachReadingCallback(readingId);}} detachReadingCallback={(readingId)=>{this.handleDetachAttachReadingCallback(readingId);}} />

              <div className="text-right bottom-btn-div">
                {(!this.props.journalData || this.userId == this.journalUserId) &&
                  <button type="submit" className="btn btn-info loginButton" disabled={this.state.isWriting || !(this.state.journalDate.length>0) || !(this.state.isDateCorrect) || this.state.isEmptyReading}>{this.props.journalData ? "Update" : "Submit"}</button>
                }
                {(this.props.journalData && this.userId == this.journalUserId) &&
                  <button onClick={()=>{this.handleDelete()}} type="button" className="btn btn-danger loginButton">Delete</button>
                }
                <button onClick={()=>{this.handleCancel()}} type="button" className="btn btn-nomal loginButton">Cancel</button>
              </div>
          </form>

        </div>
    );
  }


}
AddJournal.propTypes={
  journalData: PropTypes.object,
  history: PropTypes.object
};
export default AddJournal;
