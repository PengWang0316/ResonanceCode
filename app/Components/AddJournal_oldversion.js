import React, { Component } from "react";
import QueryString from "query-string";
import PropTypes from "prop-types";
import Util from "../apis/Util";
// import api from "../apis/api";
import DatabaseApi from "../apis/DatabaseApi";
import Loading from "./Loading";
import LoginApi from "../apis/LoginApi";

class AddJournal_oldversion extends Component{

  componentWillMount(){

    if(this.props.location){
      let queryInfo = QueryString.parse(this.props.location.search);
      this.readingId = queryInfo.readingId;
      this.readingName = queryInfo.readingName;
      this.readingDate = Util.getDateString(queryInfo.readingDate);
      // console.log("AddJournal paper readingId, readingName, and readingDate:", `Id: ${this.readingId} Name: ${this.readingName} Date: ${this.readingDate}`);
    }
    this.userId=LoginApi.isLogin(document).userid;

    let journalData = this.props.journalData;
    if(journalData){
      this.journalId=journalData.journalId;
      // this.readingDate=journalData.readingDate;
      this.readingName=journalData.readingName;
    }
    // console.log("AddJournal page:",journalData);
    // setting state
    this.state={
      isWriting: false,
      journalDate: journalData?Util.getDateString(journalData.date):Util.getCurrentDateString(),
      pingPongState: journalData?journalData.pingPongState:"Neutral",
      overview: journalData?journalData.overview:"",
      internalEnvironment: journalData?journalData.internalEnvironment:"",
      relationalEnvironment: journalData?journalData.relationalEnvironment:"",
      physicalEnvironment: journalData?journalData.physicalEnvironment:"",
      creativeVector: journalData?journalData.creativeVector:"",
      synchronicities: journalData?journalData.synchronicities:"",
      dreams: journalData?journalData.dreams:"",
      journalId: journalData?journalData.journalId:"",
      userId: journalData?journalData.userId:"",
      readingId: journalData?journalData.readingId:""
    };
    // if(this.journalId) this.setState(this.state);
  }

  componentDidMount(){
    // console.log("did mount");
    if(this.journalId) this.setState(this.state);
  }

  handleChange(event, element){
    // console.log(event.target.value);
    let newState={};
    newState[element]=event.target.value;
    this.setState(newState);
    // console.log(newState);
  }

  handleSubmit(event){
    event.preventDefault();
    // console.log("**********submit************");

      this.setState({isWriting: true});
      // console.log("submit:",this.state);
      // assemble object for creating
      let journalObject = {
        user_id: this.userid,
        reading_id: this.readingId?this.readingId:this.state.readingId,
        date: new Date(this.state.journalDate),
        overview: this.state.overview,
        internal_environment: this.state.internalEnvironment,
        relational_environment: this.state.relationalEnvironment,
        physical_environment: this.state.physicalEnvironment,
        creative_vector: this.state.creativeVector,
        synchronicities: this.state.synchronicities,
        dreams: this.state.dreams,
        ping_pong_state:this.state.pingPongState
      };
      // console.log("journalId", this.journalId);
      if(this.journalId) DatabaseApi.updateJournal(journalObject, this.state.journalId).then((reault)=>{this.props.history.push("/reading")});
      else DatabaseApi.createJournal(journalObject).then((reault)=>{this.props.history.push("/reading")});

  }

  handleDelete(event){
    event.preventDefault();
    // console.log("**********delete************");
    DatabaseApi.deleteJournal(this.state.readingId, this.state.journalId).then((reault)=>{this.props.history.push("/reading")});
  }

  render(){
    let titleDiv;
    if (this.readingId) titleDiv = <div><span className="rcTitle">Add a new journal to your reading: </span>{this.readingName}  ({this.readingDate})</div>;
      else titleDiv = <div><span className="rcTitle">Update {this.readingName}</span></div>;
    return(
      <div className="addReadingDiv">
        {this.state.isWriting && <Loading text="Creating" />}

        {titleDiv}
        <form className="form-horizontal" onSubmit={(event) => {this.handleSubmit(event);}}>

          <div className="text-right bottom-btn-div">{(!this.readingId && this.state.userId==this.userId) && <button type="button" className="btn btn-danger btn-del" onClick={(event)=>{this.handleDelete(event);}}>Delete</button>}{(this.readingId || this.state.userId==this.userId) && <button type="submit" className="btn btn-info loginButton" disabled={this.state.isWriting || !(this.state.journalDate.length>0) || !(this.state.overview.length>0)}>{this.readingId?"Submit":"Update"}</button>}</div>

            <div className="form-group row form-div">
              <label htmlFor="journalDate" className="col-sm-1 col-form-label">Date</label>
              <div className="col-sm-4">
                <input className="form-control" type="text" value={this.state.journalDate} id="journalDate" onChange={(event)=>{this.handleChange(event,"journalDate");}} />
              </div>
              <label htmlFor="pingPongState" className="col-sm-3 col-form-label">Ping Pong State</label>
              <div className="col-sm-4">
                <select value={this.state.pingPongState} id="pingPongState" className="form-control" onChange={(event)=>{this.handleChange(event,"pingPongState");}}>
                  <option value="Neutral">Neutral</option>
                  <option value="Reading/Call">Reading/Call</option>
                  <option value="Impact absorbing">Impact absorbing</option>
                  <option value="Impact complete">Impact complete</option>
                  <option value="Response">Response</option>
                  <option value="Outward moving">Outward moving</option>
                  <option value="Outward complete">Outward complete</option>
                </select>
              </div>
            </div>

            <div className="form-group form-div">
              <label htmlFor="overview" className="col-form-label">Overview</label>
              <div>
                <textarea className="form-control" rows="3" type="text" value={this.state.overview} placeholder="Overview..." id="overview" onChange={(event)=>{this.handleChange(event,"overview");}} />
              </div>
            </div>

            <div className="form-group form-div">
              <label htmlFor="internalEnvironment" className="col-form-label">Internal Environment</label>
              <div>
                <textarea className="form-control" rows="3" type="text" value={this.state.internalEnvironment} placeholder="Internal Environment..." id="internalEnvironment" onChange={(event)=>{this.handleChange(event,"internalEnvironment");}} />
              </div>
            </div>

            <div className="form-group form-div">
              <label htmlFor="relationalEnvironment" className="col-form-label">Relational Environment</label>
              <div>
                <textarea className="form-control" rows="3" type="text" value={this.state.relationalEnvironment} placeholder="Relational Environment..." id="relationalEnvironment" onChange={(event)=>{this.handleChange(event,"relationalEnvironment");}} />
              </div>
            </div>

            <div className="form-group form-div">
              <label htmlFor="physicalEnvironment" className="col-form-label">Physical Environment</label>
              <div>
                <textarea className="form-control" rows="3" type="text" value={this.state.physicalEnvironment} placeholder="Physical Environment..." id="physicalEnvironment" onChange={(event)=>{this.handleChange(event,"physicalEnvironment");}} />
              </div>
            </div>

            <div className="form-group form-div">
              <label htmlFor="creativeVector" className="col-form-label">Creative Vector</label>
              <div>
                <textarea className="form-control" rows="3" type="text" value={this.state.creativeVector} placeholder="Creative Vector..." id="creativeVector" onChange={(event)=>{this.handleChange(event,"creativeVector");}} />
              </div>
            </div>

            <div className="form-group form-div">
              <label htmlFor="synchronicities" className="col-form-label">Synchronicities</label>
              <div>
                <textarea className="form-control" rows="3" type="text" value={this.state.synchronicities} placeholder="Synchronicities..." id="synchronicities" onChange={(event)=>{this.handleChange(event,"synchronicities");}} />
              </div>
            </div>

            <div className="form-group form-div">
              <label htmlFor="dreams" className="col-form-label">Dreams</label>
              <div>
                <textarea className="form-control" rows="3" type="text" value={this.state.dreams} placeholder="Dreams..." id="dreams" onChange={(event)=>{this.handleChange(event,"dreams");}} />
              </div>
            </div>

          <div className="text-right bottom-btn-div">{(!this.readingId && this.state.userId==this.userId) && <button className="btn btn-danger btn-del" onClick={(event)=>{this.handleDelete(event);}}>Delete</button>}{(this.readingId || this.state.userId==this.userId) && <button type="submit" className="btn btn-info loginButton" disabled={this.state.isWriting || !(this.state.journalDate.length>0) || !(this.state.overview.length>0)}>{this.readingId?"Submit":"Update"}</button>}</div>

        </form>

      </div>
    );
  }

}
AddJournal_oldversion.propTypes={
  journalData: PropTypes.object
};
export default AddJournal_oldversion;
