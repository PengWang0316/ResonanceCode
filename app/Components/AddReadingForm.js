import React, { Component } from 'react';
import { connect } from 'react-redux';
// import PropTypes from 'prop-types';

import HexagramLine from './HexagramLine';
import { getCurrentDateString } from '../apis/Util';

const jQuery = require('jquery');

window.jQuery = jQuery;
require('../resources/jquery-ui.min.css');
require('../resources/jquery-ui.min');
/** The add reading form component.
 * @returns {null} No return;
*/
export class AddReadingForm extends Component {
  /*
  static propTypes = {
    readings: PropTypes.arrayOf(PropTypes.shape(PropTypes.oneOf([
      PropTypes.number,
      PropTypes.string,
      PropTypes.arrayOf(PropTypes.string),
      PropTypes.arrayOf(PropTypes.shape(PropTypes.oneOf([
        PropTypes.string,
        PropTypes.bool
      ]))),
      PropTypes.shape(PropTypes.oneOf([
        PropTypes.number,
        PropTypes.string
      ]))
    ])))
  };
  static defaultProps = {
    readings: []
  }; */
  /** Offering some states for the component.
   * @returns {object} Return a object for the initial state.
  */
  static getInitialState = () => ({
    readingName: '',
    people: '',
    date: getCurrentDateString()
  });

  state = Object.assign({}, AddReadingForm.getInitialState());

  /** Setting some state for the component.
   * @returns {null} No return.
  */
  // componentWillMount() {
  //   this.initialState();
  // }

  /** Setting up a datepicker for the Date input.
    * @returns {null} No return.
  */
  componentDidMount() {
    // Setting up datepicker
    jQuery('#date').datepicker({
      onSelect: dateText => this.setState({ date: dateText })
    });
  }

  /** Setting some state for the component.
   * @param {object} nextProps contains value for next props.
   * @returns {null} No return.
  */
  componentWillReceiveProps(nextProps) {
    if (this.props.readings.length !== 0 &&
      this.props.readings.length !== nextProps.readings.length)
      this.setState(Object.assign({}, AddReadingForm.getInitialState()));
  }

  /** Handle submit function.
   * @param {object} event is the object comes from submit a form.
   * @returns {null} No return;
  */
  handleSubmit = event => {
    event.preventDefault();
    jQuery('button[type=submit]').attr('disabled', 'disabled');
    this.props.handleSubmit(this.state);
  }

  /** Controlling the input value for the form.
   * @param {object} event is the object comes from a input's value changing.
   * @param {string} name is the input's id.
   * @returns {null} No return;
  */
  handleInputChange = event => this.setState({ [event.target.id]: event.target.value });

  /** Call the container's method when a user click a coin icon.
    * @param {int} lineNumber is the number of line.
    * @param {int} coinsPoint is the point that moment this line has based on the calculation.
    * @returns {null} No return.
  */
  handleCoinClickCallback = (lineNumber, coinsPoint) =>
    this.props.handleCoinClick(lineNumber, coinsPoint);

  /** Calling the container's cancel method when a user clicks the cancel button.
    * @returns {null} No return.
  */
  handleCancelCallback = () => this.props.handleCancel();

  /** Render the component.
   * @returns {null} No return;
  */
  render() {
    return (
      <div>
        <div className="rcTitle">Create A New Reading</div>
        <form className="form-horizontal" onSubmit={this.handleSubmit}>
          <div className="text-right bottom-btn-div">
            <button type="submit" className="btn btn-info loginButton" disabled={!(this.props.addReadingTempState.availableArr[6] && this.state.people.length > 0 && this.state.readingName.length > 0 && this.state.date.length > 0 && !this.props.addReadingTempState.isLoading)}>Submit</button>
            <button type="button" className="btn btn-normal loginButton" onClick={this.handleCancelCallback}>Cancel</button>
          </div>

          <div className="form-group row form-div">
            <label htmlFor="readingName" className="col-lg-3 col-form-label">Reading Name</label>
            <div className="col-lg-9">
              <input className="form-control" type="text" placeholder="The name for this reading" id="readingName" value={this.state.readingName} onChange={this.handleInputChange} />
            </div>
          </div>

          <div className="form-group row form-div">
            <label htmlFor="people" className="col-lg-1 col-form-label">People</label>
            <div className="col-lg-7">
              <input className="form-control" type="text" placeholder="Who is doing this with you" id="people" value={this.state.people} onChange={this.handleInputChange} />
            </div>
            <label htmlFor="date" className="col-lg-1 col-form-label">Date</label>
            <div className="col-lg-3">
              <input className="form-control" type="text" id="date" value={this.state.date} onChange={this.handleInputChange} />
            </div>
          </div>


          {/* The results of coins */}
          <div className="rcTitle coinDiv">Throw your coins and record here.</div>

          <div className="row">
            <div className="col-lg-6">
              <div className="addreading_image_title">The first image:</div>
              {this.props.addReadingTempState.availableArr[5] ? <HexagramLine lineNumber="5" side={this.props.addReadingTempState.line5.side1} middle={this.props.addReadingTempState.line5.middle1} handleCoinClick={this.handleCoinClickCallback} isFirst /> : <div className="noAvailableDiv text-center">Line 6 has not been entered.</div>}

              {this.props.addReadingTempState.availableArr[4] ? <HexagramLine lineNumber="4" side={this.props.addReadingTempState.line4.side1} middle={this.props.addReadingTempState.line4.middle1} handleCoinClick={this.handleCoinClickCallback} isFirst /> : <div className="noAvailableDiv text-center">Line 5 has not been entered.</div>}

              {this.props.addReadingTempState.availableArr[3] ? <HexagramLine lineNumber="3" side={this.props.addReadingTempState.line3.side1} middle={this.props.addReadingTempState.line3.middle1} handleCoinClick={this.handleCoinClickCallback} isFirst /> : <div className="noAvailableDiv text-center">Line 4 has not been entered.</div>}

              {this.props.addReadingTempState.availableArr[2] ? <HexagramLine lineNumber="2" side={this.props.addReadingTempState.line2.side1} middle={this.props.addReadingTempState.line2.middle1} handleCoinClick={this.handleCoinClickCallback} isFirst /> : <div className="noAvailableDiv text-center">Line 3 has not been entered.</div>}

              {this.props.addReadingTempState.availableArr[1] ? <HexagramLine lineNumber="1" side={this.props.addReadingTempState.line1.side1} middle={this.props.addReadingTempState.line1.middle1} handleCoinClick={this.handleCoinClickCallback} isFirst /> : <div className="noAvailableDiv text-center">Line 2 has not been entered.</div>}

              <HexagramLine lineNumber="0" side={this.props.addReadingTempState.line0.side1} middle={this.props.addReadingTempState.line0.middle1} handleCoinClick={this.handleCoinClickCallback} isFirst />
            </div>

            <div className="col-lg-6">
              <div className="addreading_image_title">The second image:</div>
              {this.props.addReadingTempState.line5.side2 === '' ? <div className="availableDiv text-center" >Line 6</div> : <HexagramLine lineNumber="5" side={this.props.addReadingTempState.line5.side2} middle={this.props.addReadingTempState.line5.middle2} isFirst={false} />}

              {this.props.addReadingTempState.line4.side2 === '' ? <div className="availableDiv text-center">Line 5</div> : <HexagramLine lineNumber="4" side={this.props.addReadingTempState.line4.side2} middle={this.props.addReadingTempState.line4.middle2} isFirst={false} />}

              {this.props.addReadingTempState.line3.side2 === '' ? <div className="availableDiv text-center">Line 4</div> : <HexagramLine lineNumber="3" side={this.props.addReadingTempState.line3.side2} middle={this.props.addReadingTempState.line3.middle2} isFirst={false} />}

              {this.props.addReadingTempState.line2.side2 === '' ? <div className="availableDiv text-center">Line 3</div> : <HexagramLine lineNumber="0" side={this.props.addReadingTempState.line2.side2} middle={this.props.addReadingTempState.line2.middle2} isFirst={false} />}

              {this.props.addReadingTempState.line1.side2 === '' ? <div className="availableDiv text-center">Line 2</div> : <HexagramLine lineNumber="0" side={this.props.addReadingTempState.line1.side2} middle={this.props.addReadingTempState.line1.middle2} isFirst={false} />}

              {this.props.addReadingTempState.line0.side2 === '' ? <div className="availableDiv text-center">Line 1</div> : <HexagramLine lineNumber="0" side={this.props.addReadingTempState.line0.side2} middle={this.props.addReadingTempState.line0.middle2} isFirst={false} />}
            </div>


          </div>
          <div className="change-line-div"><span>Change Lines: </span>{this.props.addReadingTempState.changeLines}</div>


          <div className="text-left bottom-btn-div">
            <button type="submit" className="btn btn-info loginButton" disabled={!(this.props.addReadingTempState.availableArr[6] && this.state.people.length > 0 && this.state.readingName.length > 0 && this.state.date.length > 0 && !this.props.addReadingTempState.isLoading)}>Submit</button>
            <button type="button" className="btn btn-normal loginButton" onClick={this.handleCancelCallback}>Cancel</button>
          </div>
        </form>
      </div>
    );
  }
}
/*
AddReadingForm.propTypes = {
  addReadingTempState: PropTypes.object.isRequired,
  handleCancel: PropTypes.func.isRequired,
  handleCoinClick: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired
}; */
const mapStateToProps = state => ({
  readings: state.readings
});
export default connect(mapStateToProps, null)(AddReadingForm);
