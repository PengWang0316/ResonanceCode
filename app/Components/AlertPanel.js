import React from 'react';

import styles from '../styles/AlertPanel.module.css';

const AlertPanel = ({
  id = '', type = 'success', children = '', style = {}
}) => (
  <div id={id} className={`alert alert-${type} mt-3 ${styles.alertPanel}`} style={style} role="alert">
    {children}
  </div>
);
export default AlertPanel;
