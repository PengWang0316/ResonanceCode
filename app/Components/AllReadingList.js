import React from 'react';
import { connect } from 'react-redux';

import Pagination from './SharedComponents/Pagination';
import LoadingAnimation from './SharedComponents/LoadingAnimation';
import { fetchAllReadingList as fetchAllReading } from '../actions/ReadingActions';
import { NUMBER_OF_READING_PER_PAGE } from '../config';
import ReadingListRow from './ReadingListRow';

/* Show the all reading in the list */
export const AllReadingList = ({
  allReadingList, readingsAmount, fetchAllReadingList, handleClick, isLoading
}) => (
  <div className="modal fade" id="readingListModal" tabIndex="-1" role="dialog" aria-labelledby="readingListModalLabel" aria-hidden="true">
    <div className="modal-dialog" role="document">
      <div className="modal-content">
        <div className="modal-header">
          <h5 className="modal-title" id="readingListModalLabel">Select a reading for this journal</h5>
          <button type="button" className="close" data-dismiss="modal" aria-label="Close">
            <span aria-hidden="true">&times;</span>
          </button>
        </div>
        <div className="modal-body">
          <LoadingAnimation />
          {!isLoading &&
            <div className="mb-3">
              {allReadingList.map(reading => <ReadingListRow key={`${reading.name}_${reading.date}`} reading={reading} handleClick={handleClick} />)}
            </div>
          }

          {/* Pagination componentWillMount
            * Keeping readingsAmount, numberPerpage, and fetchContent out of AllReadingListPagination component can help it raising reusability.
          */}
          {readingsAmount !== null && <Pagination
            amount={readingsAmount}
            fetchContent={fetchAllReadingList}
            numberPerpage={NUMBER_OF_READING_PER_PAGE}
          />}

        </div>
        <div className="modal-footer">
          <button type="button" className="btn btn-secondary" data-dismiss="modal">Close</button>
        </div>
      </div>
    </div>
  </div>
);
const mapStateToProps = state => ({
  allReadingList: state.allReadingList,
  readingsAmount: state.readingsAmount,
  isLoading: state.isLoading
});
const mapDispatchToProps = dispatch => ({
  fetchAllReadingList: pageNumber => dispatch(fetchAllReading(pageNumber))
});
// export AllReadingList;
export default connect(mapStateToProps, mapDispatchToProps)(AllReadingList);
