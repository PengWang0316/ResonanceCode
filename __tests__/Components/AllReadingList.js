import React from 'react';
import renderer from 'react-test-renderer';

import { AllReadingList } from '../../app/Components/AllReadingList';

jest.mock('../../app/Components/SharedComponents/Pagination', () => 'Pagination');
jest.mock('../../app/Components/SharedComponents/LoadingAnimation', () => 'LoadingAnimation');
jest.mock('../../app/Components/ReadingListRow', () => 'ReadingListRow');

describe('AllReadingList test', () => {
  const props = {
    allReadingList: [{ name: 'testName', date: '2018-01-01' }],
    readingsAmount: 1,
    fetchAllReadingList: jest.fn(),
    handleClick: jest.fn(),
    isLoading: false
  };

  test('AllReadingList snapshot', () => expect(renderer.create(<AllReadingList {...props} />).toJSON()).toMatchSnapshot());
});
