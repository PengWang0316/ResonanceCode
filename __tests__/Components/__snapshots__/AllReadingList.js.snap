// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllReadingList test AllReadingList snapshot 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="readingListModalLabel"
  className="modal fade"
  id="readingListModal"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
          id="readingListModalLabel"
        >
          Select a reading for this journal
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <LoadingAnimation />
        <div
          className="mb-3"
        >
          <ReadingListRow
            handleClick={[MockFunction]}
            reading={
              Object {
                "date": "2018-01-01",
                "name": "testName",
              }
            }
          />
        </div>
        <Pagination
          amount={1}
          fetchContent={Array []}
          numberPerpage={10}
        />
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-secondary"
          data-dismiss="modal"
          type="button"
        >
          Close
        </button>
      </div>
    </div>
  </div>
</div>
`;
